### PR TITLE
DDF-1769 Bugfix for preferences modal color picker.

### PR DESCRIPTION
**Pull request by shane-voisard**
@jlcsmith 
@jaymcnallie 
@bdeining  hero

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/494" target="_blank">ReviewNinja

File: catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js [hunk 1]
```diff
@@ -99,8 +99,55 @@ define([
...
+                    // mimic bootstrap styling.
+                    $replacer.css('width', '100%');
+                    $replacer.find('div.sp-dd').css('float', 'right');
+                    $replacer.find('div.sp-preview').css('width', '97%');
+                });
+                colorTabView.spReplacers = $colorInputs.siblings('div.sp-replacer');
+            },
             onShow: function () {
-                require(['spectrum']);
+                require(['spectrum'], this.initSpectrum);
```
Comment (stustison) [inline catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js]: This seems weird to me. Why wouldn't you require in spectrum with everything else being required into this view?

**Comment by stustison** [comment]
@codice/ui

File: catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js [hunk 2]
```diff
@@ -99,8 +99,55 @@ define([
...
+                var $colorInputs = $('div[id="colorTab"] input[type="color"]');
+                var colorTabView = this;
+
+                var colorChange = function (spectrumTinyColor) {
+                    var newColor = spectrumTinyColor.toHexString();
+                    this.$colorInput.val(newColor);
+                };
+
+                $colorInputs.each(function () {
+                    var $color = $(this);
```
Comment (pklinef) [inline catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js]: You should be able to add two parameters to the callback function and this value I think will be set in the second parameter.

http://api.jquery.com/jQuery.each/

File: catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js [hunk 3]
```diff
@@ -99,8 +99,55 @@ define([
...
+                    $color.spectrum({
+                        change: _.bind(colorChange, {$colorInput: $color})
+                    });
+
+                    var $replacer = $color.siblings('div.sp-replacer');
+
+                    // mimic bootstrap styling.
+                    $replacer.css('width', '100%');
+                    $replacer.find('div.sp-dd').css('float', 'right');
```
Comment (pklinef) [inline catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js]: Is there a reason this cannot be done in the style sheets?

**Comment by shane-voisard** [comment]
@stustison - removed the require('spectrum')
@pklinef - moved styling into css.

**Comment by bdeining** [comment]
Hero build successful.  Verified that the color picker renders on IE11 (Windows VM), Chrome and FireFox and that it still behaves appropriately (persistence, switching colors, reset to defaults).  LGTM.

**Pull request merged**